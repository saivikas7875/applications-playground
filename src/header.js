import React from 'react';
import AppBar from '@material-ui/core/AppBar';

export const Header = () => (
    <header className="app-header">
        <AppBar color="primary">
            <div>'Direct Flights'</div>
        </AppBar>
    </header>
);
