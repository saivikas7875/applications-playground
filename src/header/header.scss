.app-header {
    font-size: 24px;
    padding: 10px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .header-content-wrapper {
        align-items: center;
        display: flex;
        justify-content: space-between;
        width: 1600px;
        margin: auto;

        @media (max-width: 1600px) {
            width: 1400px;
        }

        @media (max-width: 1400px) {
            width: 1200px;
        }

        @media (max-width: 1200px) {
            width: 1100px;
        }
    }

    .header-content {
        align-items: center;
        display: flex;
        justify-content: space-between;
    }

    .title {
        font-size: 30px;
    }

    .app-navigation {
        display: flex;
        justify-content: space-between;
    }

    .app-navigation,
    .menu-gutter {
        padding: 0 !important;
    }

    .home-menu-item {
        padding: 6px !important;

        &:hover {
            background-color: #009688 !important;
        }
    }

    .home-icon {
        &:hover {
            background-color: #009688 !important;
        }

        svg {
            border-bottom: 5px solid transparent;

            &:hover {
                border-bottom-color: #ff3d00;
            }
        }
    }
}
