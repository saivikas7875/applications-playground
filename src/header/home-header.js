import './home-header.scss';
import React, {useState} from 'react';
import github from '../github.png';
import twitter from '../twitter.png';
import Typography from '@material-ui/core/Typography';
import gmail from '../gmail.png';

export const HomeHeader = () => {
    const [showContactCard, toggleContactCard] = useState(false);

    const style = showContactCard ? 'grid-with-contact-card' : 'initial-grid';

    return (
        <header className='home-app-header'>
            <div className={`title-container ${style}`}>
                <div className='title'>
                    <span>Applications Playground</span>
                </div>
                <div className='card-container'>
                    <div className='card-wrapper'>
                        <div className='contact-card'>
                            <div className='author-info'>
                                <div
                                    className='author'
                                    onClick={() => toggleContactCard(false)}
                                    onTouchStart={() => toggleContactCard(true)}
                                >
                                    <span className='name'>Hey there, my name is Sai Vikas Gopal.</span>
                                    <span className='caption'>Welcome to my application playground</span>
                                </div>
                                <div className='links'>
                                    <a href='https://github.com/saivikas7875' target='_blank' rel='noopener noreferrer'>
                                        <img src={github} alt='Github link' className='social-icon' />
                                    </a>
                                    <a href='https://twitter.com/saivikas_g' target='_blank' rel='noopener noreferrer'>
                                        <img src={twitter} alt='Twitter link' className='social-icon' />
                                    </a>
                                    <a href='mailto:saivikas7875@gmail.com' target='_blank' rel='noopener noreferrer'>
                                        <img src={gmail} alt='Email me' className='social-icon' />
                                    </a>
                                </div>
                            </div>
                        </div>
                        <Typography
                            className='show-information'
                            onClick={() => toggleContactCard(true)}
                            onTouchStart={() => toggleContactCard(true)}
                            aria-expanded={showContactCard}
                            aria-label='Show more'
                        >
                            Click me!
                        </Typography>
                    </div>
                </div>
            </div>
        </header>
    );
};
