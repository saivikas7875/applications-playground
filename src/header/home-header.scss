.home-app-header {
    display: flex;
    flex: 1 0 40%;
    background: black;

    .title-container {
        flex: 1 1 auto;
        align-self: center;
        display: grid;
        height: 100%;
        grid-template-columns: 1fr;
        grid-template-rows: 40% 60%;
        grid-column-gap: 20px;
        grid-template-areas:
            'title'
            'card';
    }

    .social-icon {
        height: 32px;
        width: 32px;
    }

    .card-container {
        align-self: center;
        justify-self: center;
    }

    .author-info {
        display: grid;
        grid-template-rows: 50% 50%;
        text-align: center;
        height: 100%;
        grid-template-areas:
            'author'
            'links';

        .author {
            align-items: center;
            cursor: pointer;
            display: flex;
            flex-direction: column;
            justify-content: center;
            grid-area: author;

            span + svg {
                margin-left: 10px;
            }
        }

        .links {
            align-items: center;
            grid-area: links;
            display: flex;
            justify-content: center;

            a + a {
                margin-left: 10px;
            }
        }
    }

    .card-container,
    .contact-card,
    .show-information {
        width: 300px;
        height: 100px;

        @media (max-width: 800px) {
            width: 250px;
        }
    }

    .contact-card {
        grid-area: card;
        transform: rotateY(180deg);
        background-color: white;
        box-shadow: 0px 0px 1px 3px #a7a7a7;
    }

    .contact-card,
    .show-information {
        backface-visibility: hidden;
        position: absolute;
        top: 0;
        left: 0;
    }

    .grid-with-contact-card {
        .card-wrapper {
            transform: rotateY(180deg);
        }
    }

    .card-wrapper {
        transition: 1.5s;
        transform-style: preserve-3d;
        position: relative;
    }

    .show-information {
        color: white;
        cursor: pointer;
        font-size: 16px;
        font-style: italic;
        text-align: center;
        text-decoration: underline;
        transform: rotateY(0deg);
        z-index: 2;
    }

    .title {
        align-self: center;
        justify-self: center;
        grid-area: title;
        font-size: 40px;
        color: white;
        font-weight: bold;

        @media (max-width: 1200px) {
            font-size: 32px;
        }

        @media (max-width: 1100px) {
            font-size: 28px;
        }

        @media (max-width: 900px) {
            font-size: 24px;
        }
    }
}
