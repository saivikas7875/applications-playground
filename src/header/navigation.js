import React from 'react';
import {Link} from 'react-router-dom';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import Home from '@material-ui/icons/Home';
import Button from '@material-ui/core/Button';

export const Navigation = () => {
    return (
        <MenuList className='app-navigation'>
            <MenuItem classes={{root: 'home-menu-item'}}>
                <Button component={Link} to='/' classes={{root: 'home-icon'}}>
                    <Home color='secondary' fontSize='large' />
                </Button>
            </MenuItem>
        </MenuList>
    );
};
