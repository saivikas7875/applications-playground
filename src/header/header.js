import './header.scss';
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import country from '../country.png';

export const Header = () => {
  return (
    <AppBar color="primary" classes={{ root: 'app-header' }}>
      <div className="header-content">
        <div className="title">Countries</div>
        <img src={country} className="App-logo" alt="logo" />
      </div>
    </AppBar>
  );
};
