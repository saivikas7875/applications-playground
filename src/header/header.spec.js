import {Header} from './header';
import React from 'react';
import {shallow} from 'enzyme';

describe('Application header snapshot', () => {
    let testProps = Object.freeze({
        application: ''
    });

    it('should match its previous snapshot', () => {
        expect(shallow(<Header {...testProps} />)).toMatchSnapshot();
    });
});
