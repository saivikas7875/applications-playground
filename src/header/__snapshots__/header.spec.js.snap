// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application header snapshot should match its previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    application=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="header-content-wrapper"
      >
        <div
          className="header-content"
        >
          <img
            alt="logo"
            className="App-logo"
            src="country.png"
          />
          <div
            className="title"
          >
            
          </div>
        </div>
        <WithStyles(Button)
          classes={
            Object {
              "root": "home-icon",
            }
          }
          component={[Function]}
          to="/"
        >
          <pure(HomeIcon)
            color="secondary"
            fontSize="large"
          />
        </WithStyles(Button)>
      </div>,
      "classes": Object {
        "root": "app-header",
      },
      "color": "primary",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-content"
          >
            <img
              alt="logo"
              className="App-logo"
              src="country.png"
            />
            <div
              className="title"
            >
              
            </div>
          </div>,
          <WithStyles(Button)
            classes={
              Object {
                "root": "home-icon",
              }
            }
            component={[Function]}
            to="/"
          >
            <pure(HomeIcon)
              color="secondary"
              fontSize="large"
            />
          </WithStyles(Button)>,
        ],
        "className": "header-content-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="country.png"
              />,
              <div
                className="title"
              >
                
              </div>,
            ],
            "className": "header-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "country.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "title",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(HomeIcon)
              color="secondary"
              fontSize="large"
            />,
            "classes": Object {
              "root": "home-icon",
            },
            "component": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "secondary",
              "fontSize": "large",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="header-content-wrapper"
        >
          <div
            className="header-content"
          >
            <img
              alt="logo"
              className="App-logo"
              src="country.png"
            />
            <div
              className="title"
            >
              
            </div>
          </div>
          <WithStyles(Button)
            classes={
              Object {
                "root": "home-icon",
              }
            }
            component={[Function]}
            to="/"
          >
            <pure(HomeIcon)
              color="secondary"
              fontSize="large"
            />
          </WithStyles(Button)>
        </div>,
        "classes": Object {
          "root": "app-header",
        },
        "color": "primary",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-content"
            >
              <img
                alt="logo"
                className="App-logo"
                src="country.png"
              />
              <div
                className="title"
              >
                
              </div>
            </div>,
            <WithStyles(Button)
              classes={
                Object {
                  "root": "home-icon",
                }
              }
              component={[Function]}
              to="/"
            >
              <pure(HomeIcon)
                color="secondary"
                fontSize="large"
              />
            </WithStyles(Button)>,
          ],
          "className": "header-content-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="App-logo"
                  src="country.png"
                />,
                <div
                  className="title"
                >
                  
                </div>,
              ],
              "className": "header-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "App-logo",
                  "src": "country.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "title",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(HomeIcon)
                color="secondary"
                fontSize="large"
              />,
              "classes": Object {
                "root": "home-icon",
              },
              "component": [Function],
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "secondary",
                "fontSize": "large",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
